src/components/datepicker/js/calendar.js

```diff
@@ -57,7 +57,8 @@
       scope: {
         minDate: '=mdMinDate',
         maxDate: '=mdMaxDate',
-        dateFilter: '=mdDateFilter'
+        dateFilter: '=mdDateFilter',
+        _currentView: '@mdCurrentView'
```

Comment:
Hmm, I can't really understand why this is resetted. Isn't the `currentView` method only called after some interactions?